analyzer/modules/evm.go

```diff
@@ -36,6 +37,63 @@ type EVMBlockTokenData struct {
 	TokenData map[string]*EVMTokenData
 }
 
+type EVMDeterministicError struct {
+	// Note: .error is the implementation of .Error, .Unwrap etc. It is not
+	// in the Unwrap chain. Use something like
+	// `EVMDeterministicError{fmt.Errorf("...: %w", err)}` to set up an
+	// instance with `err` in the Unwrap chain.
+	error
+}
+
+func (err EVMDeterministicError) Is(target error) bool {
+	if _, ok := target.(EVMDeterministicError); ok {
+		return true
+	}
+	return false
+}
+
+// TODO: can we move this to oasis-sdk/client-sdk/go/modules/evm?
+const EVMModuleName = "evm"
+
+var (
+	ErrEVMExecutionFailed = errors.New(EVMModuleName, 2, "execution failed")
```

Comment:
I would prefer to encode just the error codes (2, 8) as constants, and then check something like `if err.Code() == ErrCodeEVMExecutionFailed` or similar. I feel the EVM API is obligated to keep the error codes unchanged across versions, but not so for the exact wording of the error messages. Also, if my code pointer in the previous comment is correct, it seems very possible that the error message might sometimes be of the form `"execution failed: foo bar"`.

Reply:
more details on this: the oasis coded error system looks at module name and code only. see `errors.FromCode`. the grpc system calls into that to get the actual, registered error object (wrapped if it contains extra context info)

Reply:
also, indeed they do come in that form. from our tests:

> query that runs out of gas should fail: runtime client evm simulate call: execution failed: out of gas

> querying an unsupported method should fail: runtime client evm simulate call: reverted: no revert reason